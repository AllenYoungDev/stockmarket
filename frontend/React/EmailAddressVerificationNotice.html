<html>
<head>
<style>
label:after { content: ':'; }
.top-bar {margin: 0; padding: 1% 0.5% 1% 0; background: rgb(0,0,0); color: rgb(255,255,255); text-align: left; font-family: Arial, Helvetica, sans-serif;}

.email-verification-controls-section {background-color: #737CA1;color: rgb(250,250,250);}
.email-verification-controls-section-heading {text-align: left; font-size: 300%; font-family: Arial, Helvetica, sans-serif; font-weight: bold; margin: 0;}

.email-verification-controls-div {font-size: 125%; font-family: Tahoma;text-align: center;}

@media only screen and (orientation: landscape) {
.email-address-verification-email-resend-button { margin: 0.5% 0 1.5% 0; padding: 0.5%; font-size: 150%; font-weight: 900; background-color: #2B547E; border-radius: 0px; color: rgb(250,250,250); }
.email-address-verification-email-resend-button-error-notice {text-align: center; margin: 1% 0 0 0;font-size: 120%;color: rgb(70,0,0);}

}
@media only screen and (orientation: portrait) {
.email-address-verification-email-resend-button { margin: 2% 0 2% 0; padding: 0.5%; font-size: 200%; font-weight: 900; background-color: #2B547E; border-radius: 0px; color: rgb(250,250,250); }
.email-address-verification-email-resend-button-error-notice {text-align: center; margin: 1% 0 0 0;font-size: 120%;color: rgb(70,0,0);}
}


.footer   {margin: 0; padding: 0.25% 0.5% 0.1% 0; background: rgb(0,0,0); color: rgb(255,255,255); font-family: Arial, Helvetica, sans-serif; text-align: center;}
</style>
</head>
<body style="margin: 0; background-color: #FFFFFF;color: rgb(250,250,250);">

<div id="email-address-verification-notice-page"></div>

<script src="https://unpkg.com/react@18/umd/react.development.js" crossorigin></script>
<script src="https://unpkg.com/react-dom@18/umd/react-dom.development.js" crossorigin></script>

<script type="module" src="EmailAddressVerificationNotice.render.js"></script>

<noscript>Your browser must support JavaScript.</noscript>
</body>
</html>