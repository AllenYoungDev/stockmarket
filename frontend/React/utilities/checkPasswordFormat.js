const reLetterChecker = /[a-zA-Z]/;
const reNumberChecker = /[0-9]/;
const reSpecialCharacterChecker = /[\~\`\!\@\#\$\%\^\&\*\-\_\+\=\|\\\/\?\;\:\"\'\<\>\,\.\{\}\[\]\(\)]/;

export function checkPasswordFormat(password) {
    var numberOfCharactersInPassword = password.trim().length;

    if (numberOfCharactersInPassword < 10) {return false;}

    if (password.match(reLetterChecker) === null) {return false;}

    if (password.match(reNumberChecker) === null) {return false;}

    if (password.match(reSpecialCharacterChecker) === null) {return false;}

    return true;
}