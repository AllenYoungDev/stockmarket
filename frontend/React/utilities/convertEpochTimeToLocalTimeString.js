export function convertEpochTimeToLocalTimeString(epochTime) {
    var date = new Date(epochTime);
    return date.toLocaleDateString("default") + " " + date.toLocaleTimeString("default");
}