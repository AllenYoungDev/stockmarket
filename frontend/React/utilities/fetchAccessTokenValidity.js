'use strict';

import '../settings/ServerUrls.js';

export function fetchAccessTokenValidity(setLoginState, ignore) {

    console.log('fetchAccessTokenValidity() start.')

    fetch(window.backendServerUrl + "/CheckAccessTokenValidity", {method: "GET", credentials: 'include'})
    .then((response) => response.text())
    .then((text) => {
        if (text === 'valid') {
            console.log('fetchAccessTokenValidity():  access token is valid.');
            if (ignore) {return;}
            if (document.cookie.includes('admin=true')) {
                setLoginState('admin-login');
            } else {
                setLoginState('non-admin-login');
            }
        } else {
            console.log('fetchAccessTokenValidity():  access token is invalid.');
            if (ignore) {return;}
            setLoginState('no-login');
        }
    })
    .catch((err) => {
        console.log(`fetchAccessTokenValidity():  fetch() network error.  ${err}`);
        if (ignore) {return;}
        setLoginState('fetch-error');
    });
}