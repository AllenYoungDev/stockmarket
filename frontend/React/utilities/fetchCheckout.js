import '../settings/ServerUrls.js';

export function fetchCheckout(numberOfSharesToBuy, errorNoticeDomNodeRef) {
  let checkoutResponse = 0;
  let checkoutResponseStatus = 0;

  fetch(window.backendServerUrl + "/Checkout", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },      
    body: JSON.stringify({'numberOfSharesToBuy': numberOfSharesToBuy}),
    credentials: 'include',
    })
  .then((response) => {
    checkoutResponse = response;
    response.headers.forEach((value, key) => {
      console.log(`CheckoutButton() handleOnClick() checkoutResponse.headers key==>value:  ${key} ==> ${value}`)
    });
    checkoutResponseStatus=response.status;
    return response.text();})
  .then((text) => {
    if(checkoutResponseStatus === 200) {
      //window.location.href = window.frontendServerUrl;
      //This is not the proper implementation.  The page must navigate to /Checkout URL.
    } else {
      errorNoticeDomNodeRef.current.innerHTML = `Error.  Cannot proceed to checkout.  ${text}.`
    }
  })
  .catch((err) => {
    //'fetch-error'
    errorNoticeDomNodeRef.current.innerHTML = `Unable to proceed to checkout.  Network error.`;
  });   
}