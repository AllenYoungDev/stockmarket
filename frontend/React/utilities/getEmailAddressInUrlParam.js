export function getEmailAddressInUrlParam() {
    var url_string = window.location; 
    var url = new URL(url_string);
    var emailAddress = url.searchParams.get("emailAddress");
    console.log(`EmailVerificationControls() emailAddress:  ${emailAddress}.`);
    return emailAddress;
}