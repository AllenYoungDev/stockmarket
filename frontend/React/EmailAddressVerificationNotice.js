'use strict';

//import React, { StrictMode } from "react";
//import { createRoot } from 'react-dom/client';

import './utilities/SendEmailAddressVerificationEmail.js';
import { getEmailAddressInUrlParam } from './utilities/getEmailAddressInUrlParam.js';
import { footer } from './footer.js';

/* *************************************************************************************************************
EmailAddressVerificationNoticePage React component
************************************************************************************************************** */
export function EmailAddressVerificationNoticePage() {
  return React.createElement(
    React.Fragment,
    null,
    React.createElement(
      'div',
      { className: 'top-bar' },
      React.createElement(
        'a',
        { href: window.frontendServerUrl, style: { textDecoration: 'none', color: 'rgb(255,255,255)' } },
        'Allen Young\'s Stockmarket Demo'
      )
    ),
    React.createElement(
      'section',
      { className: 'email-verification-controls-section' },
      React.createElement(
        'h1',
        { className: 'email-verification-controls-section-heading' },
        'Verify Your Email Address'
      ),
      React.createElement(EmailVerificationControls, null)
    ),
    footer()
  );
}

/* *************************************************************************************************************
EmailVerificationControls React component
************************************************************************************************************** */

export function EmailVerificationControls() {
  var emailAddress = getEmailAddressInUrlParam();

  function handleOnClick(event) {
    event.stopPropagation();
    window.SendEmailAddressVerificationEmail(null, emailAddress, '');
  }

  return React.createElement(
    React.Fragment,
    null,
    React.createElement(
      'div',
      { className: 'email-verification-controls-div' },
      React.createElement(
        'p',
        null,
        'Email address:  ',
        emailAddress
      ),
      React.createElement(
        'p',
        null,
        'You cannot log in until you verify your email address.'
      ),
      React.createElement(
        'p',
        { style: { marginBottom: "0.5%" } },
        'Please check your email inbox and click the verification link to verify your email address.'
      ),
      React.createElement(
        'div',
        { style: { textAlign: "center" } },
        React.createElement('p', { id: 'error_notice', className: 'email-address-verification-email-resend-button-error-notice' }),
        React.createElement(
          'button',
          { type: 'button', className: 'email-address-verification-email-resend-button', onClick: handleOnClick },
          'Resend email-address verification email'
        )
      )
    )
  );
}