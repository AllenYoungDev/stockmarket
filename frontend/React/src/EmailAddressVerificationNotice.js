'use strict';

//import React, { StrictMode } from "react";
//import { createRoot } from 'react-dom/client';

import './settings/ServerUrls.js';
import './utilities/SendEmailAddressVerificationEmail.js';
import {getEmailAddressInUrlParam} from './utilities/getEmailAddressInUrlParam.js';
import {footer} from './footer.js';

/* *************************************************************************************************************
EmailAddressVerificationNoticePage React component
************************************************************************************************************** */
export function EmailAddressVerificationNoticePage() {
  return (
    <React.Fragment>
    <div className="top-bar">
    <a href={window.frontendServerUrl} style={{textDecoration: 'none', color: 'rgb(255,255,255)'}}>Allen Young's Stockmarket Demo</a>
    </div>
    
    <section className="email-verification-controls-section">
    <h1 className="email-verification-controls-section-heading">Verify Your Email Address</h1>
    <EmailVerificationControls />
    </section>

    {footer()}
    </React.Fragment>
  );
}

/* *************************************************************************************************************
EmailVerificationControls React component
************************************************************************************************************** */

export function EmailVerificationControls() {
  var emailAddress = getEmailAddressInUrlParam();
  
  function handleOnClick(event) {
    event.stopPropagation();
    window.SendEmailAddressVerificationEmail(null, emailAddress, '')
  }

  return (
    <React.Fragment>
    <div className="email-verification-controls-div">
    <p>Email address:  {emailAddress}</p>

    <p>You cannot log in until you verify your email address.</p>
    <p style={{marginBottom: "0.5%"}}>Please check your email inbox and click the verification link to verify your email address.</p>

    <div style={{textAlign: "center"}}>
    <p id="error_notice" className="email-address-verification-email-resend-button-error-notice"></p>
    <button type="button" className="email-address-verification-email-resend-button" onClick={handleOnClick}>Resend email-address verification email</button>
    </div>
    </div>
    </React.Fragment>
  );
}
