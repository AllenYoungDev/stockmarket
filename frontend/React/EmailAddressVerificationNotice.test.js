/**
 * @jest-environment jsdom
 */

//import React from "react";
//import ReactDOM from 'react-dom/client';

import renderer from 'react-test-renderer';
import {EmailAddressVerificationNoticePage} from './EmailAddressVerificationNotice.js';

jest.mock('./utilities/getEmailAddressInUrlParam');

import {getEmailAddressInUrlParam} from './utilities/getEmailAddressInUrlParam';

(async () => {
  if (typeof React === 'undefined') {
    globalThis.React = await import("react");
  }
  if (typeof ReactDOM === 'undefined') {
    globalThis.ReactDOM = await import("react-dom/client");
  }
})();

it('React app Jest snapshot testing for Allen Young\'s Stockmarket Demo Email Address Verification Notice Page', async () => {

  getEmailAddressInUrlParam.mockImplementation(() => {
    console.log('getEmailAddressInUrlParam() mock called!');
    return 'test@allenyoung.dev';
  });

  let testRenderer;
  let tree;

  //renderer.create() is wrapped in renderer.act() for executing useEffect() calls in <HomePage />.
  //https://github.com/facebook/react/issues/15321
  testRenderer = renderer.create(
    <EmailAddressVerificationNoticePage />,
  );

  tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();
});