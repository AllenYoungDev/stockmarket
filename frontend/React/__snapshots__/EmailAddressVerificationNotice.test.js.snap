// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React app Jest snapshot testing for Allen Young's Stockmarket Demo Email Address Verification Notice Page 1`] = `
[
  <div
    className="top-bar"
  >
    <a
      href="http://localhost"
      style={
        {
          "color": "rgb(255,255,255)",
          "textDecoration": "none",
        }
      }
    >
      Allen Young's Stockmarket Demo
    </a>
  </div>,
  <section
    className="email-verification-controls-section"
  >
    <h1
      className="email-verification-controls-section-heading"
    >
      Verify Your Email Address
    </h1>
    <div
      className="email-verification-controls-div"
    >
      <p>
        Email address:  
        test@allenyoung.dev
      </p>
      <p>
        You cannot log in until you verify your email address.
      </p>
      <p
        style={
          {
            "marginBottom": "0.5%",
          }
        }
      >
        Please check your email inbox and click the verification link to verify your email address.
      </p>
      <div
        style={
          {
            "textAlign": "center",
          }
        }
      >
        <p
          className="email-address-verification-email-resend-button-error-notice"
          id="error_notice"
        />
        <button
          className="email-address-verification-email-resend-button"
          onClick={[Function]}
          type="button"
        >
          Resend email-address verification email
        </button>
      </div>
    </div>
  </section>,
  <footer
    className="footer"
  >
    <p>
      Allen Young's Stockmarket Demo by Allen Young.
    </p>
    <p>
      For demonstrating Allen Young's full-stack web app development skills to potential employers.
    </p>
    <p>
      For more of Allen Young's software development skills demonstration, visit 
      <a
        href="https://AllenYoung.dev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        AllenYoung.dev
      </a>
       and 
      <a
        href="https://GitHub.com/AllenYoungDev"
        style={
          {
            "color": "white",
          }
        }
        target="_blank"
      >
        GitHub.com/AllenYoungDev
      </a>
      .
    </p>
    <p>
      Â© 2023 Allen Young.  All rights reserved.
    </p>
  </footer>,
]
`;
